### PR TITLE
Log when Magento is in maintenance mode

### PR DESCRIPTION
**Pull request by Ethan3600**
### Description

Logging when Magento is in maintenance mode. 

Not sure if I'm the only one that fell for this, but I've (embarrassingly) spent hours trying figuring out why Magento was throwing a 500 error; only to find that it was just in maintenance mode...

Logging this will save developers some time, and eliminate some of the guess work involved with troubleshooting obscure errors.

This is usually applicable when a deployment goes a little haywire.

    ==> var/log/system.log <==
    [2018-07-15 19:38:12] main.ERROR: Unable to proceed: the maintenance mode is enabled.

----

As a "side-affect" we'd also be logging errors coming from the [launch process](https://github.com/magento/magento2/blob/2.2-develop/lib/internal/Magento/Framework/App/Bootstrap.php#L252L258), which I don't think is that much of an issue.

### Manual testing scenarios

1. Enable maintenance mode
2. Run `tail -f var/log/system.log`
3. Expect "Unable to proceed: the maintenance mode is enabled"

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=16840) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @Ethan3600. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by DanielRuf** [comment]
Please sign the CLA.

**Comment by Ethan3600** [comment]
Signed

File: lib/internal/Magento/Framework/App/Bootstrap.php [hunk 1]
```diff
@@ -259,7 +259,7 @@ public function run(AppInterface $application)
                 \Magento\Framework\Profiler::stop('magento');
             } catch (\Exception $e) {
                 \Magento\Framework\Profiler::stop('magento');
-                $this->objectManager->get(LoggerInterface::class)->error($e);
+                $this->objectManager->get(LoggerInterface::class)->error($e->getMessage());
```
Comment (Ethan3600) [inline lib/internal/Magento/Framework/App/Bootstrap.php]: 👍

**Comment by magento-engcom-team** [comment]
Hi @Ethan3600. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.6. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**